src/java.base/share/native/launcher/main.c

```diff
@@ -167,6 +185,56 @@ static void setup_sighandler() {
     }
 }
 
+static int set_last_pid(int pid) {
+#ifdef LINUX
+    char buf[11]; // enough for int32
+    const int len = snprintf(buf, sizeof(buf), "%d", pid);
+    if (0 > len || sizeof(buf) < (size_t)len) {
+        return EINVAL;
+    }
+    const char *last_pid_filename = "/proc/sys/kernel/ns_last_pid";
+    const int last_pid_file = open(last_pid_filename, O_WRONLY|O_TRUNC, 0666);
+    if (0 > last_pid_file) {
+        return errno;
+    }
+    int res = 0;
+    if (len > write(last_pid_file, buf, len)) {
+        res = errno;
+    }
+    close(last_pid_file);
+    return res;
+#else
+    return EPERM;
+#endif
+}
+
+static void spin_last_pid(int pid) {
+    const int MaxSpinCount = pid < 1000 ? 1000 : pid;
+    for (int child = fork(), prev = 0, cnt = MaxSpinCount; child < pid; child = fork(), --cnt) {
```

Comment:
Without wait, the child will occupy the PID number forever, without any need for that. So that's wrong. May be we can simplify the loop, so we won't leak the PID?

Reply:
Done